src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2328,18 +2315,18 @@ ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
         uct_ib_mlx5dv_check_dm_ksm_reg(md);
     }
 
-    /* Enable relaxed order only if we would be able to create an indirect key
-       (with offset) for strict order access */
-    uct_ib_md_parse_relaxed_order(&md->super, md_config, ksm_atomic);
-
-    uct_ib_mlx5_devx_init_flush_mr(md);
-
     /* Zero init UMR related fields, will lazy init on first use */
     md->umr.cq        = NULL;
     md->umr.qp        = NULL;
     md->umr.mkey_hash = NULL;
     ucs_list_head_init(&md->umr.mkey_pool);
 
+    /* Enable relaxed order only if we would be able to create an indirect key
+       (with offset) for strict order access */
+    uct_ib_md_parse_relaxed_order(&md->super, md_config, ksm_atomic);
+
+    uct_ib_mlx5_devx_init_flush_mr(md);
```

Comment:
there was some confusion, when asked to move "md->umr.cq = NULL;" after line 2316 i meant the line before you applied more review comments. so it would be after "md->super.vhca_id = vhca_id;"

Reply:
ok, got it, done